test/index.js

```diff
@@ -3244,3 +3244,20 @@ See https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Boolean for i
   });
 
 });
+
+suite ('interoperability', () => {
+
+  test ('Z.equals can operate on ‘Type’ values', () => {
+    eq (Z.equals ($.Number, $.Number)) (true);
+    eq (Z.equals ($.Number, $.String)) (false);
+    eq (Z.equals ($.Array ($.Number), $.Array ($.Number))) (true);
+    eq (Z.equals ($.Array ($.Number), $.Array ($.String))) (false);
+    eq (Z.equals ($.RecordType ({x: $.Number}), $.RecordType ({x: $.Number}))) (true);
+    eq (Z.equals ($.RecordType ({x: $.Number}), $.RecordType ({y: $.Number}))) (false);
+    eq (Z.equals ($.RecordType ({x: $.Number}), $.RecordType ({x: $.String}))) (false);
+    eq (Z.equals ($.NullaryType ('X') ('') (x => true), $.NullaryType ('X') ('') (x => true))) (true);
+    eq (Z.equals ($.NullaryType ('X') ('') (x => true), $.NullaryType ('Y') ('') (x => true))) (false);
+    eq (Z.equals ($.NullaryType ('X') ('') (x => true), $.NullaryType ('X') ('') (x => false))) (true);
```

Comment:
Is this something we want?

Reply:
It's documenting the fact that we don't attempt to determine function equivalence. Relying on function identity doesn't feel good, and shouldn't be necessary given that the `url` field alone should discriminate between identically named type constructors.

Reply:
I just checked my notifications and became aware of this conversation. I recently added a `Ref` type to my lib, which mainly implements `Setoid` to allow reference equality checks. Doing this in a type-safe and explicit manner is sometimes useful, e.g. for change detection of persistent data structures. Just thought it might interest you.